common/src/Model/Polyhedron_ConvexHull.h

```diff
@@ -918,14 +942,25 @@ namespace TrenchBroom {
                 boundary.push_back(h3);
                 edge->setSecondEdge(h2);
 
+                const vm::plane<T,3> plane = makePlaneFromBoundary(boundary);
                 if (it != std::end(seam)) {
-                    const auto [valid, plane] = vm::from_points(top->position(), v2->position(), v1->position());
-                    assert(valid); unused(valid);
-
                     auto* next = *it;
 
+                    const auto checkEdge = [&](const auto* e) {
+                        const auto& point = e->firstVertex()->position();
+                        
+                        // Is the point to be added inside the plane of the face being created?
+                        if (plane.point_status(point) != vm::plane_status::inside) {
+                            return false;
+                        }
+                        
+                        // Is the point to be added colinear with the first two vertices of the face being created?
+                        const bool valid = std::get<0>(vm::from_points(top->position(), point, v1->position()));
+                        return valid;
+                    };
+                    
                     // TODO use same coplanarity check as in Face::coplanar(const Face*) const ?
-                    while (it != std::end(seam) && plane.point_status(next->firstVertex()->position()) == vm::plane_status::inside) {
+                    while (it != std::end(seam) && checkEdge(next)) {
```

Comment:
>  Can you elaborate which problems you see?

No problems, I was just noting these changes in Polyhedron::weave.
It looks like the
`// Is the point to be added colinear with the first two vertices of the face being created?`
bit is new?

Reply:
That's a bug I ran into after all the changes I had made. I don't remember why it suddenly occured, but it wasn't introduced by my changes, it was just hidden previously.